src/vampir/vampir.lisp

```diff
@@ -68,3 +68,256 @@
+                    :arguments (list a)))
+
+
+(defparameter *negative32*
+  (flet ((wires (num)
+           (make-wire :var (intern (format nil "A~a" num) :keyword))))
+    (make-alias
+     :name   :negative32
+     :inputs (list :a)
+     :body
+     (list (make-bind
+            :names (append (mapcar #'wires (alexandria:iota 32))
+                           (list (make-wire :var :|()|)))
+            :value (make-application
+                    :func      :intrange32
+                    :arguments (list (make-wire :var :a))))
+           (make-wire :var :a31)))))
+
+(defun negative32 (a)
+  (make-application :func :negative32
+                    :arguments (list a)))
+
+(defparameter *nonnegative32*
+  (make-alias :name :nonnegative32
+              :inputs (list :a)
+              :body (list
+                     (make-infix :op :-
+                                 :lhs (make-constant :const 1)
+                                 :rhs (make-application :func :negative32
+                                                        :arguments (list (make-wire :var :a)))))))
+(defun nonnegative32 (a)
+  (make-application :func :negative32
+                    :arguments (list a)))
+
+(defparameter *range31*
+  (labels ((in-plus (n)
+             (if (= n 0)
+                 (make-wire :var :a0)
+                 (make-infix :op :+
+                             :lhs (in-plus (- n 1))
+                             :rhs (make-infix :op :*
+                                              :lhs (make-constant :const (expt 2 n))
+                                              :rhs (make-wire :var (intern
+                                                                    (format nil "A~a" n) :keyword)))))))
+    (labels ((in-tuple (n)
+               (if (= n 0)
+                   (make-wire :var :a0)
+                   (make-tuples :wires (list
+                                        (in-tuple (- n 1))
+                                        (make-wire :var (intern
+                                                         (format nil "A~a" n) :keyword)))))))
+      (make-alias
+       :name :range32
+       :inputs (list :a)
+       :body
+       (append (mapcar
+                (lambda (x y)
+                  (spc:make-bind :names (list (make-wire :var y)) :value
+                                 (make-application :func :fresh
+                                                   :arguments (list (make-constant :const x)))))
+                (mapcar (lambda (x) (expt 2 x))
+                        (alexandria:iota 32 :start 1))
+                (mapcar (lambda (x) (intern (format nil "A~a" x) :keyword))
+                        (alexandria:iota 31)))
+               (list (make-equality :lhs (make-wire :var :a)
+                                    :rhs (in-plus 30)))
+               (list (make-tuples :wires (list
+                                          (in-tuple 30)
+                                          (make-wire :var :|()|)))))))))
```

Comment:
anything that makes the parser spit out

```lisp
VAMPIR> (make-tuples :wires (list (in-tuple 30) (make-wire :var :|()|)))
((((((((((((((((((((#, a12), a13), a14), a15), a16), a17), a18), a19), a20), a21), a22), a23), a24), a25), a26), a27), a28), a29), a30), ())
```

is wrong.

Further you can play around and find that 

```lisp
VAMPIR> (make-tuples :wires (list (make-wire :var :a)
                                  (make-wire :var :ab) (make-wire :var :|()|)))
(a, ab, ())
```
prints rather nicely